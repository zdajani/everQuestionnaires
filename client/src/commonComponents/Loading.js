import React from 'react';
import './styles/Loading.css';

const Loading = () => (
  <div className="loading-screen">
  	<img src={'https://tools.cisco.com/security/center/Resources/images/loader.gif'} alt="loadig" id="loading"/>
  </div>
);

export default Loading;