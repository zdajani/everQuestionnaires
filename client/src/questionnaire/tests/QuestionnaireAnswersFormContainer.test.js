import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import ConnectedQuestionnaireAnswersFormContainer, { QuestionnaireAnswersFormContainer } from '../containers/QuestionnaireAnswersFormContainer';
import Loading from '../../commonComponents/Loading';
import { formattedData } from './testData';
import ConnectedAnswersFormContainer, {AnswersFormContainer} from '../../answers/containers/AnswersFormContainer';

const middlewares = [];
const mockStore = configureMockStore(middlewares);

const setup = (formattedData, isLoading) => {
  const container = shallow(<QuestionnaireAnswersFormContainer
     questionnaire={formattedData} 
     isLoading={isLoading}/>
  );
    
  return {
    container
  };
}

const connectedSetup = (store) => {
  const connectedContainer = shallow(
      <ConnectedQuestionnaireAnswersFormContainer store={store}/>
  );
  return {
    connectedContainer
  };
}

describe('Questionnaire container', () => {
  it('renders without crashing', () => {  
    const store = mockStore({
      questionnaire: {
        isLoading: false, 
        data: null,
        errorMessage: null
      },
    });
    
    const fetchQuestionnaireSpy = jest.fn();
    const match = {params: 5}
    // using mount for Full DOM rendering to make sure 
    //the component doesn't crash 
    const mountedComponent = mount(<QuestionnaireAnswersFormContainer 
      store={store} 
      isLoading={false} 
      fetchQuestionnaire={fetchQuestionnaireSpy} 
      match={match}/>);

    expect(toJson(mountedComponent)).toMatchSnapshot();
  });
  
  describe('when questionnaire data is available', () => {
    const store = mockStore({
      questionnaire: {
        isLoading: false, 
        data: formattedData,
        errorMessage: null
      },
    });
    
    it('renders self with questionnaire props', () => {   
      const { connectedContainer } = connectedSetup(store);  

      expect(connectedContainer.props().questionnaire).toEqual(formattedData);
      expect(toJson(connectedContainer)).toMatchSnapshot();
    });
    
    it('renders Questionnaire component with questionnaire props', () => {   
      const isloading = false; 
      const { container } = setup(formattedData, isloading); 
      
      expect(toJson(container.find(ConnectedAnswersFormContainer))).toMatchSnapshot();    
    });
    
    it('does not render Loading component', () => {    
      const { container } = setup(formattedData);       
      
      expect(container).not.toContainReact(<Loading />);
    });
  });
  
  describe('when loading', () => {
    const data = null;
    const isloading = true;     
    const { container } = setup(data, isloading); 
    
    it('renders loading component', () => {  
      expect(container).toContainReact(<Loading />);
    });
    
    it('does not render Questionnaire component', () => {  
      expect(container.find(AnswersFormContainer).length).toEqual(0);
    });
  });
});