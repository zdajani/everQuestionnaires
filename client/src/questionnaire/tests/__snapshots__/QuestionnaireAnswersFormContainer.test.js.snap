// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questionnaire container renders without crashing 1`] = `
<QuestionnaireAnswersFormContainer
  fetchQuestionnaire={[Function]}
  isLoading={false}
  match={
    Object {
      "params": 5,
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div>
    <Loading>
      <div
        className="loading-screen"
      >
        <img
          alt="loadig"
          id="loading"
          src="https://tools.cisco.com/security/center/Resources/images/loader.gif"
        />
      </div>
    </Loading>
  </div>
</QuestionnaireAnswersFormContainer>
`;

exports[`Questionnaire container when questionnaire data is available renders Questionnaire component with questionnaire props 1`] = `
<Connect(AnswersFormContainer)
  questionnaire={
    Object {
      "created_at": "2017-08-18T00:02:54.066Z",
      "id": 1,
      "questions": Object {
        "1": Object {
          "id": 1,
          "label": "describe ragdoll traits",
          "name": "Ragdoll traits",
        },
        "2": Object {
          "id": 2,
          "label": "describe persian traits",
          "name": "Persian traits",
        },
      },
      "title": "Cat breeds",
      "updated_at": "2017-08-18T00:02:54.066Z",
    }
  }
/>
`;

exports[`Questionnaire container when questionnaire data is available renders self with questionnaire props 1`] = `
<QuestionnaireAnswersFormContainer
  fetchQuestionnaire={[Function]}
  isLoading={false}
  questionnaire={
    Object {
      "created_at": "2017-08-18T00:02:54.066Z",
      "id": 1,
      "questions": Object {
        "1": Object {
          "id": 1,
          "label": "describe ragdoll traits",
          "name": "Ragdoll traits",
        },
        "2": Object {
          "id": 2,
          "label": "describe persian traits",
          "name": "Persian traits",
        },
      },
      "title": "Cat breeds",
      "updated_at": "2017-08-18T00:02:54.066Z",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
